src/Esprima/Scanner.cs

```diff
@@ -947,12 +955,34 @@ public Token ScanHexLiteral(int start)
             };
         }
 
+        private Token ScanBigIntLiteral(int start, ReadOnlySpan<char> number, NumberStyles style)
+        {
+            var c = number[0];
+            if (c > 7 && Character.IsHexDigit(c))
+            {
+                // ensure we get positive number
+                number = ("0" + number.ToString()).AsSpan();
```

Comment:
ToString() on spans, maybe this method should just take the original string, that had to be converted as a span.

Reply:
Span is part of read source, this is a special case when we need to materialize in order to get parsing to work correctly.